core/src/acc_text/nlg/gf/grammar/impl.clj

```diff
@@ -89,36 +93,34 @@
   (let [function-concept (some (fn [[role concept]]
                                  (when (= :function role) concept))
                                (zipmap (map ::sg/role relations) children))
-        role-map         (reduce (fn [m [{role ::sg/role {attr-name ::sg/name} ::sg/attributes} concept]]
-                                   (cond-> m
-                                     (and (not= :function role)
-                                          (some? attr-name)) (assoc (str/lower-case attr-name) (concept->name concept))))
-                                 {}
-                                 (zipmap relations children))]
+        role-map (reduce (fn [m [{role ::sg/role {attr-name ::sg/name} ::sg/attributes} concept]]
+                           (cond-> m
+                                   (and (not= :function role)
+                                        (some? attr-name)) (assoc (str/lower-case attr-name) (concept->name concept))))
+                         {}
+                         (zipmap relations children))]
     {:name   (concept->name concept)
      :params (map concept->name children)
      :body   (variants
                (for [syntax (->> (keyword value) (get amr) (:frames) (map :syntax))]
                  (interpose {:type  :operator
                              :value "++"}
-
-                            (for [{value :value pos :pos role :role} syntax]
-                              (let [literal-val {:type :literal :value value}]
-                                (cond
-                                  (and (some? role)
-                                       (contains? role-map (str/lower-case role))) {:type  :function
-                                                                                    :value (get role-map (str/lower-case role))}
-                                  (some? role)               {:type  :literal
-                                                              :value (format "{{%s}}" role)}
-                                  (and (some? function-concept)
-                                       (= pos :VERB))       {:type  :function
-                                                             :value (concept->name function-concept)}
-                                  (= pos :ADP)              literal-val
```

Comment:
this allows nil values to be passed, changed it to check if value is not nil instead